### ISSUE TITLE
Blog Entry doesn't show up after publishing

### ISSUE THREAD

**Issue by orea** [WeTeam/WeBlog#38]
Hello,

As per the subject, I've checked the database and the content is there, but the blog entry is not displayed on the page or in the Recent Posts or Categories.

It seems to be an issue with the index. After deleting the data/indexes/WeBlog folder I copied the same from our staging environment into our production environment and the blog entry showed up.

I had tried rebuilding the WeBlog Search Index from the Sitecore Control Panel, but that didn't work. Only after I physically copied over the index files did the Blog update.

Can you provide some guidance as to a permanent fix?

Thanks.

**Comment by TehWardy** [comment]
I have the same problem.
We use version 7.2 of sitecore, both sitecore and weblog were recently updated (within the last 6 months).

Our setup involves:
  1 database server
  2 web servers

Our CM and CD servers are separate but share the same DB server.

We have confirmed that normal content management based publishing within sitecore but not within weblog seems fine but publishing weblog entries is problematic at best.

I found that taking the brute force approach (see steps listed below) is about the only way for us to get blog entries posted to our live site right now ...

Publish item in content editor on CM server.
Publish item in page editor on CD server.
IIS reset both boxes.
Clear index caches on both boxes.
Login to desktop and brute force republish hole DB on CD server.

... sometimes not all of these steps are required but it appears intermittent and unpredictable. 
I am making sure to wait a considerable time after publishing to ensure it's not a queuing / scheduling issue.

There has to be a better way, our content management users don't have this level of access to the system for a start so relying on a developer to handle this at the moment is not ideal.

I have also walked through the details on ... 
   http://sitecoreblog.alexshyba.com/2011/04/search-index-troubleshooting.html

... On the basis that other items publish fine within Sitecore I think it's reasonable to assume that the platform itself is ok and that the issue is likely localised to something in the weblog code (an assumption perhaps).

Any thoughts on how to proceed would be welcome at this point?

**Comment by TehWardy** [comment]
Interesting note: 
I would expect if the issue was with indexing that linking directly to the page would still work even if it wasn't listed in menus / generated listings.
That is not the case, despite Sitecore telling me the item has been published I can only directly link to it if in edit mode (page editor), normal viewing returns a 404.
Checking the database, the item is actually there ... hmmm.

**Comment by adeneys** [comment]
The fact that you can link to a post when editing but not view it under normal conditions indicates the item has not been published, or at least the server isn't seeing the item as published. You also listed a number of steps which generally help solve the issue. It sounds like a setup issue. Any of the following could be causing the issue:

* Remote events not configured correctly so CD doesn't see when CM pushes the item
* Caching applied on a rendering which uses an index to locate items
* Index not seeing the published items and not rebuilding

Start with the data issue first as the potential index issues will never be the cause if the data is the issue.

* Create a new item based on `sample/sample item` template.
* Publish it to the CM server, verify everything is working as expected (item appears in the web DB for CM)
* Publish it to CD server, verify the item appears in web DB for CD server
* Repeat above with a WeBlog post item

In the above steps, we're never looking at the website, just the content tree once logged in to verify the data is what we expect.

If the above works as expected then we can move onto the index related issues.

* Use the log file on the CD server to verify when the Index has finished updating after a publish.
* Check the blog listing page. If the new post doesn't show up try clearing the cache through the `/sitecore/admin/cache.aspx` utility.
* If clearing the cache makes the post show up then the issue is a publish / re-index race condition. You need to remove the caching on the post listing rendering.
* If that doesn't solve the issue try using the desktop on CD to rebuild the WeBlog index, then clear the cache as above.
* If that solves the issue then the problem would be that the index isn't seeing the item updates and isn't rebuilding itself.

Progress through that and see if any helps.

**Comment by TehWardy** [comment]
In our case I would not be suprised if there were setup issues, I don't spend enough time with Sitecore to know how to resolve the issues but it looks like the actions I have taken so far have resulted in at least a temporary fix.

In case this helps anyone else I found the following to be helpful ...

 - Restart IIS on both boxes (probably overkill).
 - Rebuilt all indexes on both internal and external boxes.
 - Forced full republish of everything to empty queues (probably overkill).
 - confirm with logs and databases that queues are completely flushed.
 - Updated timings in web.config files on both boxes to ensure that scheduled publishing occurs on same interval (shouldn't matter).
 - Synced clocks on the servers

Synching the clocks on the servers may have been a factor in our case ... seems to have helped.
We have a single domain controller that syncs with an outside world time server then all other servers time sync with that. Making that happen on our Sitecore boxes appears useful.

Maybe all this dependency on tasks / scheduling is causing some issues.

**Comment by TehWardy** [comment]
Ok this isn't fixed ... I found that I can work around the problem by taking a combination of the actions above but I don't want our content editors depending on me every time they want to publish a blog entry. 

From what I can tell, iisreset's appear to be the solving step ... it's like maybe the server is caching that it's done something it actually hasn't done and reloading the whole site brings it to realise that.

I've been trying to get my head round the underlying data as documentation is very vague with SC databases this is a bit of an issue ...

http://stackoverflow.com/questions/28875188/sitecore-publishing-problems-and-determining-item-state/28878402?noredirect=1#comment46055603_28878402

Odd that we only get this problem with weblog items though ...
Is there something I can do to force weblog items to basically behave like normal SC items?

**Comment by alan-null** [comment]
A lot changed since then. I don't think this issue is still valid. If it is, please create separate issue.

**Comment by TehWardy** [comment]
Easier to just not use this crap if i have to check your bugs every 6 months.
